### PR TITLE
Add Neural Code Search Dataset

### PR DESCRIPTION
**Pull request by vinaykudari**

### CODE REVIEW

File: datasets/neural_code_search/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+size_categories:
+  evaluation_dataset:
+  - n<1K
+  search_corpus:
+  - n>1M
+source_datasets:
+- original
+task_categories:
+- question-answering
+task_ids: []
```
Comment (lhoestq) [inline datasets/neural_code_search/README.md:21]: could you add a task id here ?
If no task id matches in the tagger you can just put a generic description of the task like
```
task_ids:
- other-my-generic-task-description
```

File: datasets/neural_code_search/neural_code_search.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                            "question_author_url": row["question_author_url"],
+                            "answer": row["answer"],
+                            "answer_url": row["answer_url"],
+                            "answer_author": row["answer_author"],
+                            "answer_author_url": row["answer_author_url"],
+                            "examples": row["examples"],
+                            "examples_url": row["examples_url"],
+                        }
+            else:
+                for dirpath, _, fnames in os.walk(dp):
```
Comment (lhoestq) [inline datasets/neural_code_search/neural_code_search.py:156]:
```suggestion
                for dirpath, _, fnames in sorted(os.walk(dp)):
```
we need to make the order deterministic here

File: datasets/neural_code_search/neural_code_search.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                    for fname in fnames:
```
Comment (lhoestq) [inline datasets/neural_code_search/neural_code_search.py:157]:
```suggestion
                    for fname in sorted(fnames):
```
same here

**Review by lhoestq — COMMENTED** [review]
Really good thanks !

I left a few comments

**Comment by vinaykudari** [comment]
> Really good thanks !
> 
> I left a few comments

Thanks, resolved them :)

**Comment by lhoestq** [comment]
looks like this PR includes changes about many other files than the ones for Code Search

can you create another branch and another PR please ?

**Comment by vinaykudari** [comment]
> looks like this PR includes changes about many other files than the ones for Code Search
> 
> can you create another branch and another PR please ?

Okay sure

**Pull request closed without merging**